src/lib/zf/log.c

```diff
@@ -135,14 +136,44 @@ void zf_log(struct zf_stack* st, const char* fmt, ...)
   va_end(va);
 }
 
+static int zf_log_set_file_name(const char *file)
+{
+  if( strnlen(file, ZF_LOG_FILE_NAME_SIZE) > ZF_LOG_FILE_NAME_SIZE-1 ) {
+    zf_log_stack_err(NO_STACK, "%s: File name is longer than %i characters: %s\n",
+                     __func__, ZF_LOG_FILE_NAME_SIZE-1, strerror(-EINVAL));
+    return -EINVAL;
+  }
+
+  strncpy(zf_log_file_name, file, sizeof(zf_log_file_name));
+  zf_log_file_name[ZF_LOG_FILE_NAME_SIZE-1] = '\0';
```

Comment:
```suggestion

```
strncpy will null terminate the destination so there's no need for this.

Reply:
thanks, and the correct size is checked just above, so it's redundant.

Reply:
included and tested.